File: TsinghuaNet.Helper/Models/NetCredential.vb
Imports System.Net.Http
Imports MvvmHelpers

Public Class NetCredential
    Inherits ObservableObject

    Private _State As NetState
    Public Property State As NetState
        Get
            Return _State
        End Get
        Set(value As NetState)
            SetProperty(_State, value)
        End Set
    End Property

    Private _Username As String
    Public Property Username As String
        Get
            Return _Username
        End Get
        Set(value As String)
            SetProperty(_Username, value)
        End Set
    End Property

    Private _Password As String
    Public Property Password As String
        Get
            Return _Password
        End Get
        Set(value As String)
            SetProperty(_Password, value)
        End Set
    End Property

    Private Shared ReadOnly Client As New HttpClient()

    Public Function GetHelper() As IConnect
        Return ConnectHelper.GetHelper(State, Username, Password, Client)
    End Function

    Public Function GetUseregHelper() As UseregHelper
        Return New UseregHelper(Username, Password, Client)
    End Function
End Class

File: TsinghuaNet/Model/NetCredential.cs
﻿using System.Net.Http;
using MvvmHelpers;
using TsinghuaNet.Helper;

namespace TsinghuaNet.Model
{
    public class NetCredential : ObservableObject
    {
        private NetState state;
        public NetState State
        {
            get => state;
            set => SetProperty(ref state, value);
        }

        private string username;
        public string Username
        {
            get => username;
            set => SetProperty(ref username, value);
        }

        private string password;
        public string Password
        {
            get => password;
            set => SetProperty(ref password, value);
        }

        private static readonly HttpClient Client = new HttpClient();

        public IConnect GetHelper() => ConnectHelper.GetHelper(State, Username, Password, Client);

        public UseregHelper GetUseregHelper() => new UseregHelper(Username, Password, Client);
    }
}
